File: src/main/java/com/bot/service/entity/BandService.java
package com.bot.service.entity;

import com.bot.model.entities.Band;
import com.bot.model.entities.BotUser;
import com.bot.model.repository.BandRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.telegram.telegrambots.meta.api.objects.User;


@Service
@Transactional
public class BandService {

    @Autowired
    private BandRepository bandRepository;
    @Autowired
    private CategoryService categoryService;

    public Band createNewBand(User user){
        Band newBand = new Band();
        newBand.setSingle(true);
        newBand.setAdminId(user.getId());
        return bandRepository.save(newBand);
    }

    public Band createNewBand(BotUser user){
        Band newBand = new Band();
        newBand.setSingle(true);
        newBand.setAdminId(user.getUserId());
        return bandRepository.save(newBand);
    }

    public void deleteBandWithCategories(Band band){
        band.setBandUsers(null);
        categoryService.getCategoriesByBand(band)
                .forEach(category -> categoryService.deleteCategory(category));
        bandRepository.delete(band);
    }

    public void saveBand(Band band){
        bandRepository.save(band);
    }

    public void isNotSingleAnymore(Band band){
        band.setSingle(false);
        bandRepository.save(band);
    }

    public void setSingleIfNeeded(Band band){
        if(band.getBandUsers().size() == 1){
            band.setSingle(true);
            bandRepository.save(band);
        }
    }
}

File: telegram-adapter/src/main/kotlin/com/planner/service/entity/BandService.kt
package com.planner.service.entity

import com.planner.model.entities.User
import com.planner.model.entities.UserGroup
import com.planner.repository.GroupRepository
import org.springframework.stereotype.Service

@Service
class BandService(
    private val groupRepository: GroupRepository
) {
    fun createNewGroup(user: User) =
        UserGroup()
            .apply {
                admin = user
                users = setOf(user)
            }
            .let { groupRepository.save(it) }

//    fun createNewBand(user: TelegramUser): UserGroup {
//        val newBand = UserGroup()
//        newBand.setSingle(true)
//        newBand.setAdminId(user.getUserId())
//        return groupRepository.save(newBand)
//    }

    fun deleteBandWithCategories(band: UserGroup) {
        groupRepository.delete(band)
    }

    fun saveBand(band: UserGroup) {
        groupRepository.save(band)
    }
}